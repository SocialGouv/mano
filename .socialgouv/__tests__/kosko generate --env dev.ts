//

import { getEnvManifests } from "@socialgouv/kosko-charts/testing";
import { project } from "@socialgouv/kosko-charts/testing/fake/github-actions.env";

jest.setTimeout(1000 * 60);
test("kosko generate --dev", async () => {
  process.env.HARBOR_PROJECT = "mano";
  expect(
    await getEnvManifests("dev", "", {
      ...project("mano").dev,
      KUBE_NAMESPACE: "mano-186-master-dev2",
      //RANCHER_PROJECT_ID: "c-bd7z2:p-dtlsm",
    })
  ).toMatchSnapshot();
});
