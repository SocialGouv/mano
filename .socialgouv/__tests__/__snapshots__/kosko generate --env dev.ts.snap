// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kosko generate --dev 1`] = `
"---
apiVersion: v1
kind: Namespace
metadata:
  annotations:
    socialgouv/creator: autodevops
    janitor/ttl: 7d
    field.cattle.io/creatorId: gitlab
    field.cattle.io/projectId: c-gjtkk:p-hb8lf
    git/branch: refs/heads/mybranch
    git/remote: socialgouv/mano
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/mano
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    azure-pg-admin-user: mano
    application: mano
    component: mano
    owner: mano
    team: mano
    cert: wildcard
  name: mano-mybranch
---
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    kapp.k14s.io/delete-strategy: orphan
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/mano
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: api
    application: mano
    component: mano
    owner: mano
    team: mano
    cert: wildcard
  name: api
  namespace: mano-mybranch
spec:
  replicas: 1
  selector:
    matchLabels:
      app: api
  template:
    metadata:
      annotations:
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        kapp.k14s.io/delete-strategy: orphan
        app.github.com/job: '5678'
        app.github.com/ref: refs/heads/mybranch
        app.github.com/repo: socialgouv/mano
        app.github.com/run: '1234'
        app.github.com/sha: '0123456'
      labels:
        app: api
        application: mano
        component: mano
        owner: mano
        team: mano
        cert: wildcard
    spec:
      containers:
        - image: >-
            ghcr.io/socialgouv/mano/api:sha-0123456789abcdefghijklmnopqrstuvwxyz0123
          livenessProbe:
            failureThreshold: 6
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 30
            periodSeconds: 5
            timeoutSeconds: 5
          name: api
          ports:
            - containerPort: 3000
              name: http
          readinessProbe:
            failureThreshold: 15
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 0
            periodSeconds: 5
            successThreshold: 1
            timeoutSeconds: 1
          resources:
            limits:
              cpu: '1'
              memory: 1Gi
            requests:
              cpu: 250m
              memory: 256Mi
          startupProbe:
            failureThreshold: 12
            httpGet:
              path: /healthz
              port: http
            periodSeconds: 5
          volumeMounts:
            - mountPath: /mnt/files
              name: files
          envFrom:
            - secretRef:
                name: azure-pg-user-mybranch
          env:
            - name: STORAGE_DIRECTORY
              value: /mnt/files
      imagePullSecrets:
        - name: regcred
      volumes:
        - name: files
          emptyDir: {}
      initContainers:
        - env:
            - name: WAIT_FOR_RETRIES
              value: '24'
          envFrom:
            - secretRef:
                name: azure-pg-user-mybranch
          image: ghcr.io/socialgouv/docker/wait-for-postgres:6.56.1
          imagePullPolicy: Always
          name: wait-for-postgres
          resources:
            limits:
              cpu: 20m
              memory: 32Mi
            requests:
              cpu: 5m
              memory: 16Mi
        - args:
            - initdb
          command:
            - yarn
          image: >-
            ghcr.io/socialgouv/mano/api:sha-0123456789abcdefghijklmnopqrstuvwxyz0123
          imagePullPolicy: Always
          name: init
          resources:
            limits:
              cpu: 500m
              memory: 512Mi
            requests:
              cpu: 100m
              memory: 128Mi
          envFrom:
            - secretRef:
                name: azure-pg-user-mybranch
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: api
    application: mano
    component: mano
    owner: mano
    team: mano
    cert: wildcard
  name: api
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    kapp.k14s.io/delete-strategy: orphan
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/mano
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  namespace: mano-mybranch
spec:
  ports:
    - name: http
      port: 80
      targetPort: 3000
  selector:
    app: api
  type: ClusterIP
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    kapp.k14s.io/delete-strategy: orphan
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/mano
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: api
    application: mano
    component: mano
    owner: mano
    team: mano
    cert: wildcard
  name: api
  namespace: mano-mybranch
spec:
  rules:
    - host: mano-mybranch.dev2.fabrique.social.gouv.fr
      http:
        paths:
          - backend:
              service:
                name: api
                port:
                  name: http
            path: /
            pathType: Prefix
  tls:
    - hosts:
        - mano-mybranch.dev2.fabrique.social.gouv.fr
      secretName: wildcard-crt
---
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    kapp.k14s.io/delete-strategy: orphan
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/mano
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: dashboard
    application: mano
    component: mano
    owner: mano
    team: mano
    cert: wildcard
  name: dashboard
  namespace: mano-mybranch
spec:
  replicas: 1
  selector:
    matchLabels:
      app: dashboard
  template:
    metadata:
      annotations:
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        kapp.k14s.io/delete-strategy: orphan
        app.github.com/job: '5678'
        app.github.com/ref: refs/heads/mybranch
        app.github.com/repo: socialgouv/mano
        app.github.com/run: '1234'
        app.github.com/sha: '0123456'
      labels:
        app: dashboard
        application: mano
        component: mano
        owner: mano
        team: mano
        cert: wildcard
    spec:
      containers:
        - image: >-
            ghcr.io/socialgouv/mano/dashboard:sha-0123456789abcdefghijklmnopqrstuvwxyz0123
          livenessProbe:
            failureThreshold: 6
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 30
            periodSeconds: 5
            timeoutSeconds: 5
          name: dashboard
          ports:
            - containerPort: 80
              name: http
          readinessProbe:
            failureThreshold: 15
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 0
            periodSeconds: 5
            successThreshold: 1
            timeoutSeconds: 1
          resources:
            limits:
              cpu: 500m
              memory: 256Mi
            requests:
              cpu: 100m
              memory: 128Mi
          startupProbe:
            failureThreshold: 12
            httpGet:
              path: /healthz
              port: http
            periodSeconds: 5
            initialDelaySeconds: 30
      imagePullSecrets:
        - name: regcred
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: dashboard
    application: mano
    component: mano
    owner: mano
    team: mano
    cert: wildcard
  name: dashboard
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    kapp.k14s.io/delete-strategy: orphan
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/mano
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  namespace: mano-mybranch
spec:
  ports:
    - name: http
      port: 80
      targetPort: 80
  selector:
    app: dashboard
  type: ClusterIP
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    kapp.k14s.io/delete-strategy: orphan
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/mano
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: dashboard
    application: mano
    component: mano
    owner: mano
    team: mano
    cert: wildcard
  name: dashboard
  namespace: mano-mybranch
spec:
  rules:
    - host: dashboard-mano-mybranch.dev2.fabrique.social.gouv.fr
      http:
        paths:
          - backend:
              service:
                name: dashboard
                port:
                  name: http
            path: /
            pathType: Prefix
  tls:
    - hosts:
        - dashboard-mano-mybranch.dev2.fabrique.social.gouv.fr
      secretName: wildcard-crt
---
apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
metadata:
  name: netpol-mano-mybranch
  namespace: mano-mybranch
spec:
  ingress:
    - from:
        - podSelector: {}
    - from:
        - namespaceSelector:
            matchLabels:
              network-policy/source: ingress-controller
    - from:
        - namespaceSelector:
            matchLabels:
              network-policy/source: monitoring
  podSelector: {}
  policyTypes:
    - Ingress
---
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    kapp.k14s.io/delete-strategy: orphan
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/mano
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: www
    application: mano
    component: nginx
    owner: mano
    team: mano
    cert: wildcard
  name: www
  namespace: mano-mybranch
spec:
  replicas: 1
  selector:
    matchLabels:
      app: www
  template:
    metadata:
      annotations:
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        kapp.k14s.io/delete-strategy: orphan
        app.github.com/job: '5678'
        app.github.com/ref: refs/heads/mybranch
        app.github.com/repo: socialgouv/mano
        app.github.com/run: '1234'
        app.github.com/sha: '0123456'
      labels:
        app: www
        application: mano
        component: nginx
        owner: mano
        team: mano
        cert: wildcard
    spec:
      containers:
        - image: >-
            ghcr.io/socialgouv/mano/website:sha-0123456789abcdefghijklmnopqrstuvwxyz0123
          livenessProbe:
            failureThreshold: 6
            httpGet:
              path: /index.html
              port: http
            initialDelaySeconds: 30
            periodSeconds: 5
            timeoutSeconds: 5
          name: www
          ports:
            - containerPort: 80
              name: http
          readinessProbe:
            failureThreshold: 15
            httpGet:
              path: /index.html
              port: http
            initialDelaySeconds: 0
            periodSeconds: 5
            successThreshold: 1
            timeoutSeconds: 1
          resources:
            limits:
              cpu: 500m
              memory: 128Mi
            requests:
              cpu: 5m
              memory: 32Mi
          startupProbe:
            failureThreshold: 12
            httpGet:
              path: /index.html
              port: http
            periodSeconds: 5
      imagePullSecrets:
        - name: regcred
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: www
    application: mano
    component: nginx
    owner: mano
    team: mano
    cert: wildcard
  name: www
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    kapp.k14s.io/delete-strategy: orphan
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/mano
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  namespace: mano-mybranch
spec:
  ports:
    - name: http
      port: 80
      targetPort: 80
  selector:
    app: www
  type: ClusterIP
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    kapp.k14s.io/delete-strategy: orphan
    app.github.com/job: '5678'
    app.github.com/ref: refs/heads/mybranch
    app.github.com/repo: socialgouv/mano
    app.github.com/run: '1234'
    app.github.com/sha: '0123456'
  labels:
    app: www
    application: mano
    component: nginx
    owner: mano
    team: mano
    cert: wildcard
  name: www
  namespace: mano-mybranch
spec:
  rules:
    - host: app-mano-mybranch.dev2.fabrique.social.gouv.fr
      http:
        paths:
          - backend:
              service:
                name: www
                port:
                  name: http
            path: /
            pathType: Prefix
  tls:
    - hosts:
        - app-mano-mybranch.dev2.fabrique.social.gouv.fr
      secretName: wildcard-crt
"
`;
